File: zouk-sets.js
const fs = require('fs')

const rootFolder = './processed-files'
const jsonDatabase = `${rootFolder}/database.json`
const outputFileLocation = `${rootFolder}/zouk-sets.json`

const database = JSON.parse(fs.readFileSync(jsonDatabase, 'utf8'))

if (!database) {
  console.error('Database not found')
}

const songList = database.VirtualDJ_Database.Song
const zoukSets = []

for (const i in songList) {
  const song = songList[i]
  const tags = song.Tags
  if (Object.hasOwnProperty.call(tags, 'Genre') && Object.hasOwnProperty.call(tags, 'Author')) {
    const genre = tags.Genre
    const author = tags.Author

    if (genre.trim().toLowerCase() === 'zouk set' && author.trim().toLowerCase() === 'dj sprenk') {
      zoukSets.push(song)
    }
  }
}

fs.writeFile(outputFileLocation, JSON.stringify(zoukSets), {}, function (err) {
  console.error(err)
})

File: zouk-sets.py
import json

JSON_DB_FILE = "./processed-files/database.json"
OUTPUT_FILE = "./processed-files/zouk-sets.json"

# Note: keep these lower case to make case-insensitive compare work
GENRE_FILTER = "zouk set"
ARTIST_FILTER = "dj sprenk"


def load_file(file_name=JSON_DB_FILE):
    """Load the file and return the parsed data"""
    with open(file_name) as json_file:
        return json.load(json_file)


def set_filter(elem):
    """Filter function for returning a specified genre"""
    genre = elem.get("Tags", {}).get("@Genre", "")
    artist = elem.get("Tags", {}).get("@Author", "")

    return genre.lower() == GENRE_FILTER and artist.lower() == ARTIST_FILTER


def write_json_file(data, output_json_path):
    """Serialize the data dict to JSON and write to the output_json_path location"""
    # generate the object using json.dumps() corresponding to json data
    json_data = json.dumps(data)

    # Write the json data to output json file
    with open(output_json_path, "w") as json_file:
        json_file.write(json_data)
        json_file.close()


if __name__ == "__main__":
    db = load_file()
    songs = db["VirtualDJ_Database"]["Song"]
    filtered = filter(set_filter, songs)

    breakpoint()
    filtered_sets = []
    for item in filtered:
        filtered_sets.append(item)

    breakpoint()

    print(f"Identified {len(filtered_sets)} sets out of {len(songs)} entries.")

    write_json_file(filtered_sets, OUTPUT_FILE)
